edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -53,9 +54,9 @@ impl DockerModuleRuntime {
         let client = Client::builder().build(UrlConnector::new(docker_url)?);
 
         // extract base path - the bit that comes after the scheme
-        let base_path = get_base_path(docker_url);
+        let base_path = get_base_path(docker_url)?;
         let mut configuration = Configuration::new(client);
-        configuration.base_path = base_path.to_string();
+        configuration.base_path = base_path.to_str().unwrap().to_string();
```

Comment:
Can this fail? We should probably convert it to an `expect` or handle the error. I think we should prefer to propagate the error than unwrap it.

Reply:
`to_str()` will only return `None` if the value is not valid UTF-8. The value in this case is `url.path()`, which is always percent-encoded ASCII, so it will be valid UTF-8. I don't mind replacing `unwrap` with `expect`...